modules/tools/vehicle_calibration/sensor_calibration/extract_data.py

```diff
@@ -62,57 +63,47 @@ def process_dir(path, operation):
 
     return True
 
-def extract_camera_data(dest_dir, msg, ratio):
+def extract_camera_data(dest_dir, msg):
     """
     Extract camera file from message according to ratio
     """
-    time_nseconds = []
-    pre_time_second = 0
-
-    seq = 0
-    # Check timestamp.
     #TODO: change saving logic 
-    #while True:
-    if True:
-        cur_time_second = msg.timestamp
-
-        image = sensor_image_pb2.Image()
-        image.ParseFromString(msg.message)
-        # Save image according to cyber format.
-        # height = 4, image height, that is, number of rows.
-        # width = 5,  image width, that is, number of columns.
-        # encoding = 6, as string, type is 'rgb8', 'bgr8' or 'gray'.
-        # step = 7, full row length in bytes.
-        # data = 8, actual matrix data in bytes, size is (step * rows).
-        # type = CV_8UC1 if image step is equal to width as gray, CV_8UC3
-        # if step * 3 is equal to width.
-
-        if image.encoding == "rgb8" or image.encoding == "bgr8":
-            if image.step != image.width * 3:
-                print("Image.step %d does not equal Image.width %d * 3 for color image" %
-                    (image.step, image.width))
-                return False
-        elif image.encoding == "gray" or image.encoding == "y":
-            if image.step != image.width:
-                print("Image.step %d does not equal Image.width %d or gray image" %
-                    (image.step, image.width))
-                return False
-        else:
-            print("Unsupported image encoding type %s" %encoding)
+    cur_time_second = msg.timestamp
+    image = IMAGE_OBJ
```

Comment:
Is it intended to reuse the image instance every time?

Reply:
yes. create image instance every time might cause time cost. In the future, will put the functions in a class and image instance as a member.